### PR TITLE
clusterloader2: add "--from-from-cluster" flag to support in-cluster client config if "--kubeconfig" is empty

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

File: clusterloader2/cmd/clusterloader.go [hunk 1]
```diff
@@ -61,7 +60,7 @@ var (
 )
 
 func initClusterFlags() {
-	flags.StringEnvVar(&clusterLoaderConfig.ClusterConfig.KubeConfigPath, "kubeconfig", "KUBECONFIG", "", "Path to the kubeconfig file")
+	flags.StringEnvVar(&clusterLoaderConfig.ClusterConfig.KubeConfigPath, "kubeconfig", "KUBECONFIG", "", "Path to the kubeconfig file (optional, if empty, creates using in-cluster client config)")
```
Comment (wojtek-t) [inline clusterloader2/cmd/clusterloader.go:63]: I'm fine with supporting incluster config, but I think this is a bit too implicit.

How about adding an explicit "run-from-cluster" (or sth along those lines) flag and:
- if true, create in-cluster config and validate kubeconfig is unset
- if false, use kubeconfig (and validate it is set)

?

Comment (gyuho) [reply clusterloader2/cmd/clusterloader.go:63]: Good point. Will add that flag.

File: clusterloader2/cmd/clusterloader.go [hunk 2]
```diff
@@ -61,7 +60,8 @@ var (
 )
 
 func initClusterFlags() {
-	flags.StringEnvVar(&clusterLoaderConfig.ClusterConfig.KubeConfigPath, "kubeconfig", "KUBECONFIG", "", "Path to the kubeconfig file")
+	flags.StringEnvVar(&clusterLoaderConfig.ClusterConfig.KubeConfigPath, "kubeconfig", "KUBECONFIG", "", "Path to the kubeconfig file, --run-from-cluster must be false")
+	flags.BoolEnvVar(&clusterLoaderConfig.ClusterConfig.RunFromCluster, "run-from-cluster", "RUN_FROM_CLUSTER", false, "creates using in-cluster client config, --kubeconfig must be unset")
```
Comment (wojtek-t) [inline clusterloader2/cmd/clusterloader.go:64]: nit: the description text seems strange (some noun is missing; I guess you meant

`Whether to use in-cluster client-config to create a client, --kubeconfig must be unset`

File: clusterloader2/cmd/clusterloader.go [hunk 3]
```diff
@@ -83,9 +83,20 @@ func initClusterFlags() {
...
 	errList := errors.NewErrorList()
-	if clusterLoaderConfig.ClusterConfig.KubeConfigPath == "" {
-		errList.Append(fmt.Errorf("no kubeconfig path specified"))
+
+	// if '--run-from-cluster=true', create in-cluster config and validate kubeconfig is unset
+	// if '--run-from-cluster=false', use kubeconfig (and validate it is set)
+	switch clusterLoaderConfig.ClusterConfig.RunFromCluster {
+	case true:
+		if clusterLoaderConfig.ClusterConfig.KubeConfigPath != "" {
+			errList.Append(fmt.Errorf("unexpected kubeconfig path specified %q", clusterLoaderConfig.ClusterConfig.KubeConfigPath))
```
Comment (wojtek-t) [inline clusterloader2/cmd/clusterloader.go:92]: ... specified %q when --run-from-cluster is set

**Pull request merged**